src/docs/user/references/VariablesReference.adoc

```diff
@@ -1,140 +1,183 @@
 | -
+| <<_pre_post_clean, pre/post>> (not available on clean script),<<_scripting_language_support, script task>>,<<_fork_environment, environment>>,<<_control_flow_scripts,flow script>>
 
-| *User space*. See <<_data_spaces>>. Available via script task, ForkEnvironment, pre script, post script, flow script.
+| *User space*. Location of the user space. See <<_data_spaces>>.
 | user.getVirtualURI()
 | $USERSPACE
 | -
+| <<_pre_post_clean, pre/post>> (not available on clean script),<<_scripting_language_support, script task>>,<<_fork_environment, environment>>,<<_control_flow_scripts,flow script>>
 
-| *Global space*. See <<_data_spaces>>. Available via script task, ForkEnvironment, pre script, post script, flow script.
+| *Global space*. Location of the global space. See <<_data_spaces>>.
 | global.getVirtualURI()
 | $GLOBALSPACE
 | -
+| <<_pre_post_clean, pre/post>> (not available on clean script),<<_scripting_language_support, script task>>,<<_fork_environment, environment>>,<<_control_flow_scripts,flow script>>
 
-| *Input space*. See <<_data_spaces>>. Available via script task, ForkEnvironment, pre script, post script, flow script.
+| *Input space*. Location of the input space. See <<_data_spaces>>.
 | input.getVirtualURI()
 | -
 | -
+| <<_pre_post_clean, pre/post>> (not available on clean script),<<_scripting_language_support, script task>>,<<_fork_environment, environment>>,<<_control_flow_scripts,flow script>>
 
-| *Local space*. See <<_data_spaces>>. Available via script task, ForkEnvironment, pre script, post script, flow script.
+| *Local space*. Location of the local space. See <<_data_spaces>>.
 | localspace.getVirtualURI()
 | $LOCALSPACE
 | -
+| <<_pre_post_clean, pre/post>> (not available on clean script),<<_scripting_language_support, script task>>,<<_fork_environment, environment>>,<<_control_flow_scripts,flow script>>
 
-| *Output space*. See <<_data_spaces>>. Available via script task, ForkEnvironment, pre script, post script, flow script.
+| *Output space*. Location of the output space. See <<_data_spaces>>.
 | output.getVirtualURI()
 | -
 | -
+| <<_pre_post_clean, pre/post>> (not available on clean script),<<_scripting_language_support, script task>>,<<_fork_environment, environment>>,<<_control_flow_scripts,flow script>>
 
-| *Selection*. See <<_selection>>. Only available via SelectionScript.
+| *Selection*. variable which must be set to select the node. See <<_selection>>.
 | selected = true
 | -
 | -
+| <<_selection, selection>>
 
-| *Fork Environment*. Fork Environment variable is a complex java object allowing a script to set various initialization parameters of the forked JVM. Available only via EnvironmentScript.
+| *Fork Environment*. Fork Environment variable is a complex java object allowing a script to set various initialization parameters of the forked JVM. See <<_fork_environment, Fork Environment>>
 | forkEnvironment.setJavaHome( "/usr/java/default" )
 | -
 | -
+| <<_fork_environment, environment>>
```

Comment:
what do you mean by environment? environment script? we have such kind of script?